core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -858,39 +861,45 @@ public boolean load() {
 
         if (!allowWrites && dataAccessDefaultType.isMMap())
             dataAccessDefaultType = DAType.MMAP_RO;
-        if (encodingManager == null)
-            // we did not call init(), so we build the encoding manager now.
-            // just like when calling init, users have to make sure they use the same setup for import and load
-            buildEncodingManagerAndOSMParsers(flagEncodersString, encodedValuesString, dateRangeParserString, profilesByName.values());
 
         GHDirectory directory = new GHDirectory(ghLocation, dataAccessDefaultType);
         directory.configure(dataAccessConfig);
-        baseGraph = new BaseGraph.Builder(getEncodingManager())
-                .setDir(directory)
-                .set3D(hasElevation())
-                .withTurnCosts(encodingManager.needsTurnCostsSupport())
-                .setSegmentSize(defaultSegmentSize)
-                .build();
         properties = new StorableProperties(directory);
-        checkProfilesConsistency();
 
-        if (!new File(ghLocation).exists())
+        if (!new File(ghLocation).exists()) {
+            // there is nothing to load, so we create the EM from the config
+            buildEncodingManagerAndOSMParsers(flagEncodersString, encodedValuesString, dateRangeParserString, profilesByName.values());
```

Comment:
We build the encoding manager (and the parsers) from the config file, but only when there is no -gh folder.